lib/rules/prefer-query-matchers.ts

```diff
@@ -0,0 +1,108 @@
+import { TSESTree } from '@typescript-eslint/utils';
+
+import { createTestingLibraryRule } from '../create-testing-library-rule';
+import { findClosestCallNode, isMemberExpression } from '../node-utils';
+
+export const RULE_NAME = 'prefer-query-matchers';
+export type MessageIds = 'wrongQueryForMatcher';
+export type Options = [
+	{
+		validEntries: {
+			query: 'get' | 'query';
+			matcher: string;
+		}[];
+	}
+];
+
+export default createTestingLibraryRule<Options, MessageIds>({
+	name: RULE_NAME,
+	meta: {
+		docs: {
+			description:
+				'Ensure the configured `get*`/`query*` query is used with the corresponding matchers',
+			recommendedConfig: {
+				dom: 'error',
+				angular: 'error',
+				react: 'error',
+				vue: 'error',
+				marko: 'error',
+			},
+		},
+		messages: {
+			wrongQueryForMatcher: 'Use `{{ query }}By*` queries for {{ matcher }}',
+		},
+		schema: [
+			{
+				type: 'object',
+				additionalProperties: false,
+				// TODO: Is this the correct schema to use? Schema doesn't seem to be used at the lowest object level
+				properties: {
+					validEntries: {
+						type: 'array',
+						items: {
+							type: 'object',
+							properties: {
+								query: {
+									// TODO: Limit strings to the defined query types
+									type: 'string',
+								},
+								matcher: {
+									type: 'string',
+								},
+							},
+						},
+					},
+				},
+			},
+		],
+		type: 'suggestion',
+	},
+	defaultOptions: [
+		{
+			validEntries: [{ query: 'get', matcher: 'toBeVisible' }],
```

Comment:
@Belco90 I just remembered that you had suggested no query/matchers configured by default. We will be sure to make that change before we open the final PR. Putting a default here helped us get started writing the tests.